src/__a11y__/to-validate-a11y.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Axe from 'axe-core';
+import { HtmlValidate } from 'html-validate';
+import { uniq } from 'lodash';
+import { runOptions, spec } from './axe';
+
+declare global {
+  /* eslint-disable-next-line @typescript-eslint/no-namespace */
+  namespace jest {
+    /* eslint-disable-next-line @typescript-eslint/ban-types, @typescript-eslint/no-unused-vars */
+    interface Matchers<R, T = {}> {
+      toValidateA11y(): Promise<R>;
+    }
+  }
+}
+
+Axe.configure(spec);
+
+const htmlValidator = new HtmlValidate({
+  extends: ['html-validate:recommended'],
+  rules: {
+    // set relaxed to exclude error that id does not begin with letter
+    'valid-id': ['error', { relaxed: true }],
+    'no-inline-style': 'off',
+    'prefer-native-element': ['error', { exclude: ['listbox'] }],
+    // enable 'no-redundant-for' and 'element-permitted-content' after fixing Toggle
+    'no-redundant-for': 'off',
+    'element-permitted-content': 'off',
+  },
+});
+
+async function toValidateA11y(this: jest.MatcherUtils, element: HTMLElement) {
+  // Disable color-contrast checks as unavailable in unit test environment.
+  const axeResult = await Axe.run(element, { ...runOptions, rules: { 'color-contrast': { enabled: false } } });
+
+  const htmlValidateResult = htmlValidator.validateString(element.outerHTML);
```

Comment:
Not sure what is happening here. The check always passes, if there are no axe violations. html validation checks only take place if axe fails. Why is that?